utils/src/lib.rs

```diff
@@ -505,6 +542,47 @@ pub async fn fourbyte_event(topic: &str) -> Result<Vec<(String, i32)>> {
         .collect::<Vec<(String, i32)>>())
 }
 
+/// Pretty print calldata and if available, fetch possible function signatures
+///
+/// ```no_run
+/// 
+/// use foundry_utils::pretty_calldata;
+///
+/// # async fn foo() -> eyre::Result<()> {
+///   let pretty_data = pretty_calldata("0x70a08231000000000000000000000000d0074f4e6490ae3f888d1d4f7e3e43326bd3f0f5".to_string()).await?;
+///   println!("{}",pretty_data);
+/// # Ok(())
+/// # }
+/// ```
+
+pub async fn pretty_calldata(calldata: String) -> Result<PossibleSigs> {
+    let mut possible_info = PossibleSigs::new();
+    let calldata = calldata.strip_prefix("0x").unwrap();
+
+    let selector =
+        calldata.get(..8).ok_or_else(|| eyre::eyre!("calldata cannot be less that 4 bytes"))?;
+
+    let sigs: Vec<String> = fourbyte(selector).await?.into_iter().map(|sig| sig.0).collect();
+    let (_, data) = calldata.split_at(8);
+
+    if data.len() % 64 != 0 {
+        eprintln!("\nInvalid calldata size");
+        std::process::exit(0)
+    }
```

Comment:
since this is now a utility function, we shouldn't exit here and instead `eyre::bail!` here